Added doc example to explorer sortFn (#564)
jackyzha0/quartz: docs/features/explorer.md

})
```

> 

Reviewer: ```suggestion
### Use `sort` with pre-defined sort order
```

We should probably add some kind of heading for this example

> Here's another example where a map is used to define the order of the **folders** and **files** in the explorer.

Reviewer: ```suggestion
Here's another example where a map containing file/folder names or frontmatter titles is used to define the sort order of the explorer.
```

```ts title="quartz.layout.ts"
Component.Explorer({
  sortFn: (a, b) => {
    const nameOrderMap: Record<string, number> = {
      "poetry-folder": 1,
      "essay-folder": 2,
      "research-Paper-file": 3,
      "other-folder": 4,
    }

    let orderA = 0
    let orderB = 0

    if (a.file && a.file.slug) {
      orderA = nameOrderMap[a.file.slug] || 0
    } else if (a.name) {
      orderA = nameOrderMap[a.name] || 0
    }

    if (b.file && b.file.slug) {
      orderB = nameOrderMap[b.file.slug] || 0
    } else if (b.name) {
      orderB = nameOrderMap[b.name] || 0
    }

    return orderA - orderB
  },

> })

Reviewer: How are files/folders not contained in `nameOrderMap` sorted? We should probably describe that behaviour in the docs.

Author: They go to the top, I'll add it to the doc.

```

> 

Reviewer: ```suggestion
In this example, `nameOrderMap` contains slugs of files and folders with corresponding numbers that define the sort order (ascending?).
```

should probably specify if the numbers in the map sort ascending or descending and give a bit of an overview of what the code example does

Depending on your situation, you might have to use `.name`, `.displayName` or `.name.slug`

- `.name` = The default file/folder name in the explorer
- `.displayName` = The new name you mapped a file/folder name to inside of the explorer

> - `.name.slug` = The raw file/folder name

Reviewer 2: ```suggestion
- `.file.slug` = The raw file/folder name
```

Author: Oh yes of course, file.slug

> Depending on your situation, you might have to use `.name`, `.displayName` or `.name.slug`
> 
> - `.name` = The default file/folder name in the explorer
> - `.displayName` = The new name you mapped a file/folder name to inside of the explorer
> - `.file.slug` = The raw file/folder name
> 

Reviewer: ```suggestion
Instead of `Filenode.file.slug`, you could also change the example to use `Filenode.name`, if you want to use frontmatter titles (or file name if not set) inside `nameOrderMap` instead.

To make this change, swap every occurrence of `a.file.slug`/`b.file.slug` to `a.name`/`b.name` and use the frontmatter title/default file or folder name inside `nameOrderMap`
```

I don't think we should recommend using `.displayName` in official docs as those are only used for display inside the explorer and shouldn't really be used in something like a sort function (also could have side effects if people use `sort` before they use `map`)

Also, im not sure if the existing usage of `a.name`/`b.name` would need to be changed in your example if you were to use `Filenode.name` instead of `Filenode.file.slug`, would be nice if you could verify it working as intended with that change.

Author: My current setup looks like this, it might be worth mentioning, that one should give the folders numbers like 100, 200, 300, ... so that the numbers between that can be used to sort the files in the folders without having to constantly rename the order values
![image](https://github.com/jackyzha0/quartz/assets/85735034/d5fb72c7-cead-4320-a2cb-289a97027ac0)

Author: just tested it and having the `mapFn` function before or after `sortFn` doesn't matter, added it to doc

})
```

### Use `sort` with pre-defined sort order

> Here's another example where a map containing file/folder names is used to define the sort order of the explorer in quartz. All files/folders that aren't listed inside of `nameOrderMap` will appear at the top of that folders hierarchy level. It's also worth mentioning, that the smaller the number set in `nameOrderMap`, the higher up the entry will be in the explorer. Incrementing every folder/file by 100, makes ordering files in their folders a lot easier. Lastly, having the `mapFn` function before or after `sortFn` doesn't matter.

Reviewer: > Lastly, having the `mapFn` function before or after `sortFn` doesn't matter.

Might be a bit confusing for people, should probably expand to explain that using `mapFn` before or after only doesn't matter so long as only the frontmatter slugs are being used in `sort` or just not mentioned?

```ts title="quartz.layout.ts"
Component.Explorer({
  sortFn: (a, b) => {
    const nameOrderMap: Record<string, number> = {
      "poetry-folder": 100,
      "essay-folder": 200,
      "research-paper-file": 201,
      "dinosaur-fossils-file": 300,
      "other-folder": 400,
    }

    let orderA = 0
    let orderB = 0

    if (a.file && a.file.slug) {
      orderA = nameOrderMap[a.file.slug] || 0
    } else if (a.name) {
      orderA = nameOrderMap[a.name] || 0
    }

    if (b.file && b.file.slug) {
      orderB = nameOrderMap[b.file.slug] || 0
    } else if (b.name) {
      orderB = nameOrderMap[b.name] || 0
    }

    return orderA - orderB
  },
})
```

For reference, this is how the quartz explorer window would look like with that example.

```
📖 Poetry Folder
📑 Essay Folder
    ⚗️ Research Paper File
🦴 Dinosaur Fossils File
🔮 Other Folder
```

And this is how the file structure would look like.

```
index.md
poetry-folder
    index.md
essay-folder
    index.md
    research-paper-file.md
dinosaur-fossils-file.md
other-folder
    index.md
```

> As you can see `nameOrderMap` is using the raw file/folder name (without file extensions). If you would rather use the frontmatter name of the file/folder, then ... **(ToDo: how does one modify the code so the `nameOrderMap` can be filled with the frontmatter name?)**

Reviewer: There's still a remaining TODO here, is this sentence intentional? (should probably resolve TODO or remove sentence if its just a leftover)

Author: The "..." with the ToDo description was what I was talking about yesterday.

"would you be so kind and add that, or alternatively we could also leave away that last part entirely, as it's not good practice to reference frontmatter names in code anyway."
